omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,346 @@
+		Allergen.setOtherNonCodedConceptUuid(RestTestConstants2_0.ALLERGY_OTHER_NON_CODED_UUID);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_UUID; // allergy does not support uuid
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 4;
+	}
+	
+	/**
+	 *  Delete Allergy
+	 */
+	@Test
+	public void shouldDeleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertFalse(allergy.isVoided());
+		
+		// attempt to delete allergy
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	/**
+	 *  Delete Allergy incorrectly
+	 *  throw ObjectNotFoundException because allergy does not exist
+	 */
+	@Test(expected = ObjectNotFoundException.class)
+	public void shouldThrowExceptionWhenDeletingNonExistentAllergy() throws Exception
+	{
+		// attempt to delete allergy
+		handle(newDeleteRequest(getURI() + "/" + "NonExistentAllergyUUID", new Parameter("reason", "unit test")));
+	}
+
+	/**
+	 *  Delete all Allergies
+	 */
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		// attempt to delete all allergies
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/*
+	 *  Get Allergy 
+	 */
+	@Test
+	public void shouldGetAllergyByUuid() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+
+		// attempt to get allergy by uuid
+		SimpleObject so = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+
+		Assert.assertEquals(allergy.getComment(), Util.getByPath(so, "comment"));
+		Assert.assertEquals(allergy.getSeverity().getUuid(), Util.getByPath(so, "severity/uuid"));

```

Comment:
the priginal depended on the Jackson code. we want to compare the original allergy with the allergy that the api returns as saved, this is simpleObject without Jackson code to convert to allergy object.